tests/certs_test.go

```diff
@@ -1,28 +1,422 @@
+	Eventually(cmdWithRetry(curlCmd, strconv.Itoa(expectedStatusCode), maxRetryIterations)).Should(BeTrue())
+}
+
 var _ = Describe("Certs", func() {
+	var testApp App
+	var domain string
+	var certName string
+	var customSSLEndpoint string
+
+	certPath := path.Join(getDir(), "./certs")
+	certs := map[string]Cert{
+		"www": Cert{
+			Name:     "www-foo-com",
+			CertPath: fmt.Sprintf("%s/www.foo.com.cert", certPath),
+			KeyPath:  fmt.Sprintf("%s/www.foo.com.key", certPath)},
+		"wildcard": Cert{
+			Name:     "wildcard-foo-com",
+			CertPath: fmt.Sprintf("%s/wildcard.foo.com.cert", certPath),
+			KeyPath:  fmt.Sprintf("%s/wildcard.foo.com.key", certPath)},
+		"foo": Cert{
+			Name:     "foo-com",
+			CertPath: fmt.Sprintf("%s/foo.com.cert", certPath),
+			KeyPath:  fmt.Sprintf("%s/foo.com.key", certPath)},
+		"bar": Cert{
+			Name:     "bar-com",
+			CertPath: fmt.Sprintf("%s/bar.com.cert", certPath),
+			KeyPath:  fmt.Sprintf("%s/bar.com.key", certPath)},
+	}
+
+	Context("with an app yet to be deployed", func() {
+		BeforeEach(func() {
+			gitInit()
+			testApp = App{Name: getRandAppName()}
+			cmd := createApp(testApp.Name)
+			Eventually(cmd).Should(SatisfyAll(
+				Say("Git remote deis added"),
+				Say("remote available at ")))
+
+			domain = getRandDomain()
+			certName = strings.Replace(domain, ".", "-", -1)
+		})
+
+		It("can add, attach, list, and remove certs", func() {
+			addDomain(domain, testApp.Name)
+
+			addCert(certName, certs["wildcard"].CertPath, certs["wildcard"].KeyPath)
+
+			Eventually(certsInfo(certName)).Should(Say("No connected domains"))
+
+			attachCert(certName, domain)
+
+			Eventually(certsInfo(certName)).Should(Say(domain))
 
-	// Basic "Smoke" test
-	XIt("can add, list, and remove certs", func() {
-		// "deis domains:add %s --app=%s", domain, app
-		// "deis certs:list", app
-		// "deis certs:add %s %s", certPath, keyPath
-		// wait for 60 seconds until cert generation is done?
-		// curl the custom SSL endpoint
-		// "deis certs:remove %s", domain
-		// "deis certs:list", app
-		// curl the custom SSL endpoint, should fail
-		// curl app at both root and custom domain, custom should fail
-		// "deis domains:remove %s --app=%s", domain, app
+			removeCerts([]string{certName})
+		})
 	})
 
-	// See examples in https://github.com/deis/workflow/tree/master/rootfs/api/tests
-	// for other use cases perhaps needing to be tested
+	Context("with a deployed app", func() {
+		once := &sync.Once{}
+
+		BeforeEach(func() {
+			// Set up the test app only once and assume the suite will clean up.
+			once.Do(func() {
+				testApp = deployApp("example-go")
+			})
+			domain = getRandDomain()
+			certName = strings.Replace(domain, ".", "-", -1)
+			customSSLEndpoint = strings.Replace(testApp.URL, "http", "https", 1)
+		})
+
+		// Basic "Smoke" test verifying SSL endpoints
+		It("can add, attach, list, and remove certs, plus verify SSL endpoint(s)", func() {
```

Comment:
Do we believe this "smoke" scenario is still needed since we perform the same actions [here?] (https://github.com/deis/workflow-e2e/pull/75/files#diff-4e69408eafdc533cb4f2550813c5ba3cR331) (though in a slightly more complicated scenario)